src/soko_game.rs

```diff
@@ -145,12 +152,11 @@ fn handle_move(prev_level: &World, direction: Direction) -> Option<World> {
     let mut soko_box_move = None;
     if let Some((_, player_position)) = player_move.clone() {
         for (index, entity) in level.entities.iter().enumerate() {
-            if let Entity::SokoBox(soko_box) = entity {
+            if let Entity::SokoBox { position } = entity {
                 // if there is a soko_box where the player wants to move see if we can
                 // push it.
-                if soko_box.position == player_position.clone() {
-                    let new_position =
-                        get_new_position(soko_box.position.clone(), &direction);
+                if *position == player_position {
```

Comment:
can you explain what changed in the code or in your intent that introduced the dereferencing * character here? Is Coordinate typed differently somehow because of how Entity was altered?